### PR TITLE
Various cleanups to get_mult_function

### PR DESCRIPTION
**Pull request by eric-wieser**
Again, I would recommend this be reviewed commit by commit

### CODE REVIEW

File: clifford/__init__.py [hunk 1]
```diff
@@ -155,87 +157,101 @@ def construct_tables(
...
             if v_val != 0.0:
-                m = m_list_copy[ind]
+                m = m_list[ind]
                 ov_val = other_value[m]
                 if ov_val != 0.0:
-                    l = l_list_copy[ind]
-                    output[l] += v_val * mult_table_vals_copy[ind] * ov_val
+                    l = l_list[ind]
+                    output[l] += v_val * mult_table_vals[ind] * ov_val
         return output
```
Comment (eric-wieser) [inline clifford/__init__.py]: @hugohadfield: Does this actually help vs just doing the multiply by zero?

If not, this entire function can be removed.

**Comment by eric-wieser** [comment]
Last commit (f3a5a5d4281c707274764a937b2fce0b4d1af344) made travis noticeably slower, so backing it out for now

File: clifford/__init__.py [hunk 2]
```diff
@@ -160,77 +160,91 @@ def get_mult_function(k_list, l_list, m_list, mult_table_vals, n_dims, gradeList
...
-        for i in range(product_mask.shape[0]):
-            k_list_copy[i] = k_list[product_mask[i]]
-            l_list_copy[i] = l_list[product_mask[i]]
-            m_list_copy[i] = m_list[product_mask[i]]
-            mult_table_vals_copy[i] = mult_table_vals[product_mask[i]]
+    if product_mask is not None:
+        k_list = k_list[product_mask]
+        l_list = l_list[product_mask]
+        m_list = m_list[product_mask]
+        mult_table_vals = mult_table_vals[product_mask]
```
Comment (eric-wieser) [inline clifford/__init__.py]: Note that this loop is just a regular numpy indexing operation

File: clifford/__init__.py [hunk 3]
```diff
@@ -160,77 +160,91 @@ def get_mult_function(k_list, l_list, m_list, mult_table_vals, n_dims, gradeList
...
+    @numba.njit
+    def mv_mult(value, other_value):
+        output = np.zeros(n_dims)
+        for ind, k in enumerate(k_list):
+            m = m_list[ind]
+            l = l_list[ind]
+            output[l] += value[k] * mult_table_vals[ind] * other_value[m]
+        return output
+
+    return mv_mult
```
Comment (eric-wieser) [inline clifford/__init__.py]: This code just moved

File: clifford/__init__.py (hunk 1 above)
Comment (hugohadfield) [reply clifford/__init__.py]: The first check almost definitely should (if v_val != 0.0:) but the second im not so sure. Once upon a time I did make a notebook comparing these things.. I think it is worth setting up a proper benchmarking system

**Pull request merged**